plugins/statistics.koplugin/main.lua

```diff
@@ -177,14 +178,25 @@ function ReaderStatistics:addToMainMenu(tab_item_table)
                 end
             },
             {
+                text = _("Reading progress"),
+                callback = function()
+                    UIManager:setDirty(nil, "full")
```

Comment:
I see, this setDirty logic is better handled side the widget that got passed to the `UIManager:show` call. Because the widget itself knows its own size/dimension, therefore it can give an accurate refresh region to the setDirty call. For example code, you can take a look at keyvaluepage: https://github.com/koreader/koreader/blob/master/frontend/ui/widget/keyvaluepage.lua#L306.

Make sure you only call setDirty with the final calculated widget dimension.